compiler/src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -207,11 +207,14 @@ abstract class Reporter extends interfaces.ReporterResult {
       b += countString(warningCount, "warning") + " found"
     if (errorCount > 0)
       b += countString(errorCount, "error") + " found"
-    for ((settingName, count) <- unreportedWarnings)
-      b += s"there were $count ${settingName.tail} warning(s); re-run with $settingName for details"
     b.mkString("\n")
   }
 
+  def summarizeUnreportedWarnings(using Context): Unit =
+    for (settingName, count) <- unreportedWarnings do
+      val msg = s"there were $count ${settingName.tail} warning(s); re-run with $settingName for details"
```

Comment:
Here you can use the `countString` method to deal with the plural of `"warning"`.

```suggestion
      val msg = s"there were ${countString(count, settingName.tail + " warning")}; re-run with $settingName for details"
```

Reply:
Updated.